Inconsistent long overflow with a date field

### Elasticsearch Version

8.12.0

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

Elastic Cloud

### Problem Description

When posting a document, with a field definition of:
```
                "time_field" : {
                    "type" : "date",
                    "format" : "date_optional_time||epoch_millis",
                    "index" : false
                }
```

If the value is > 292278994, I get a long overflow exception (eg: `292278994` is OK, `292278995` and higher fails)

### Steps to Reproduce

Mapping:
```
"mappings": {
            "dynamic": "strict",
            "properties": {
                "@timestamp": {
                    "type": "date",
                    "format": "date_optional_time||epoch_millis"
                },
                "time_field" : {
                    "type" : "date",
                    "format" : "date_optional_time||epoch_millis",
                    "index" : false
                }
            }
        }
```

Test file:
```
cat test-file.json
{
  "@timestamp": "1708109088737",
  "time_field": 292278995
}
```


```
curl -X POST --insecure https://localhost:9200/logs-events-elasticsearchidentificationevent/_doc -u 'elastic:elastic123' -H "Content-Type: application/json"  -d @test-file.json
{"error":{"root_cause":[{"type":"document_parsing_exception","reason":"[1:58] failed to parse field [time_field] of type [date] in document with id 'aYJPwo0BtRzlmlC49I_Z'. Preview of field's value: '292278995'"}],"type":"document_parsing_exception","reason":"[1:58] failed to parse field [time_field] of type [date] in document with id 'aYJPwo0BtRzlmlC49I_Z'. Preview of field's value: '292278995'","caused_by":{"type":"arithmetic_exception","reason":"long overflow"}},"status":400}
```

Looking at the document being stored (the last successful one, with a value of `292278994`)

```
{
  "_index": ".ds-logs-events-elasticsearchidentificationevent-2024.02.19-000001",
  "_id": "aIJPwo0BtRzlmlC41Y9c",
  "_version": 1,
  "_score": 0,
  "_source": {
    "@timestamp": "1708109088737",
    "time_field": 292278994
  },
  "fields": {
    "@timestamp": [
      "2024-02-16T18:44:48.737Z"
    ],
    "time_field": [
      "+292278994-01-01T00:00:00.000Z"
    ]
  }
}
```

It seems it is parsing `292278994` as a date, however, if I change the file (quoting the number in strings)
{
  "@timestamp": "1708109088737",
  "time_field": "292278995"
}

It still fails with the same error. How can this be OK for `@timestamp` but not for `time_field`?



### Logs (if relevant)

```
[instance-0000000014/[.ds-logs-events-elasticsearchidentificationevent-2024.02.15-000026/[1/

failed to execute bulk item (create) index {[logs-events-elasticsearchidentificationevent/[elasticsearchIdentificationEvent+11+10000981/, source[n/a, actual length: [23.1kb/, max length: 2kb/} org.elasticsearch.index.mapper.DocumentParsingException: 

[1:1924/failed to parse field [time_field/of type [date/ in document with id 'elasticsearchIdentificationEvent+11+10000981'. 

Preview of field's value: '978366495' at org.elasticsearch.index.mapper.FieldMapper.rethrowAsDocumentParsingException(FieldMapper.java:232) ~

[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.FieldMapper.parse(FieldMapper.java:185) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrField(DocumentParser.java:411) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.DocumentParser.parseValue(DocumentParser.java:642) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.DocumentParser.innerParseObject(DocumentParser.java:342) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.DocumentParser.parseObjectOrNested(DocumentParser.java:293) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.DocumentParser.internalParseDocument(DocumentParser.java:144) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.DocumentParser.parseDocument(DocumentParser.java:96) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.DocumentMapper.parse(DocumentMapper.java:92) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.shard.IndexShard.prepareIndex(IndexShard.java:1012) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.shard.IndexShard.applyIndexOperation(IndexShard.java:960) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.shard.IndexShard.applyIndexOperationOnPrimary(IndexShard.java:904) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.action.bulk.TransportShardBulkAction.executeBulkItemRequest(TransportShardBulkAction.java:360) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.action.bulk.TransportShardBulkAction$2.doRun(TransportShardBulkAction.java:224) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.action.bulk.TransportShardBulkAction.performOnPrimary(TransportShardBulkAction.java:291) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.action.bulk.TransportShardBulkAction.dispatchedShardOperationOnPrimary(TransportShardBulkAction.java:142) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.action.bulk.TransportShardBulkAction.dispatchedShardOperationOnPrimary(TransportShardBulkAction.java:74) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.action.support.replication.TransportWriteAction$1.doRun(TransportWriteAction.java:216) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26) ~[elasticsearch-8.12.0.jar:?/
 at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?/
 at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?/
 at java.lang.Thread.run(Thread.java:1583) ~[?:?/
 Caused by: java.lang.ArithmeticException: long overflow at java.lang.Math.multiplyExact(Math.java:1032) ~[?:?/
 at java.lang.Math.multiplyExact(Math.java:1008) ~[?:?/
 at java.time.Instant.toEpochMilli(Instant.java:1262) ~[?:?/
 at org.elasticsearch.index.mapper.DateFieldMapper$Resolution$1.convert(DateFieldMapper.java:95) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.DateFieldMapper$DateFieldType.parse(DateFieldMapper.java:490) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.DateFieldMapper.parseCreateField(DateFieldMapper.java:910) ~[elasticsearch-8.12.0.jar:?/
 at org.elasticsearch.index.mapper.FieldMapper.parse(FieldMapper.java:183) ~[elasticsearch-8.12.0.jar:?/
```

--- a/docs/reference/mapping/params/format.asciidoc
+++ b/docs/reference/mapping/params/format.asciidoc
@@ -70,6 +70,11 @@ The following tables lists all the defaults ISO formats supported:
     (separated by `T`), is optional.
     Examples: `yyyy-MM-dd'T'HH:mm:ss.SSSZ` or  `yyyy-MM-dd`.
 
+    NOTE: When using `date_optional_time`, the parsing is lenient and will attempt to parse
+    numbers as a year (e.g. `292278994` will be parsed as a year). This can lead to unexpected results
+    when paired with a numeric focused format like `epoch_second` and `epoch_millis`.
+    It is recommended you use `strict_date_optional_time` when pairing with a numeric focused format.
+
 [[strict-date-time-nanos]]`strict_date_optional_time_nanos`::
 
     A generic ISO datetime parser, where the date must include the year at a minimum, and the time

